trillian/util/log_leaf.go

```diff
@@ -80,11 +82,11 @@ func ExtraDataForChainHash(cert ct.ASN1Cert, chainHash []byte, isPrecert bool) (
 // buildLogLeaf builds the trillian.LogLeaf. The chainHash argument controls
 // whether ExtraDataForChain or ExtraDataForChainHash method will be called.
 // If chainHash is not nil, but neither is chain, then chain will be ignored.
-func buildLogLeaf(logPrefix string, merkleLeaf ct.MerkleTreeLeaf, leafIndex int64, cert ct.ASN1Cert, chain []ct.ASN1Cert, chainHash []byte, isPrecert bool) (trillian.LogLeaf, error) {
+func buildLogLeaf(logPrefix string, merkleLeaf ct.MerkleTreeLeaf, leafIndex int64, cert ct.ASN1Cert, chain []ct.ASN1Cert, chainHash []byte, isPrecert bool) (*trillian.LogLeaf, error) {
 	leafData, err := tls.Marshal(merkleLeaf)
 	if err != nil {
 		klog.Warningf("%s: Failed to serialize Merkle leaf: %v", logPrefix, err)
-		return trillian.LogLeaf{}, err
+		return &trillian.LogLeaf{}, err
```

Comment:
`nil` is more standard than a pointer to the empty value when you return an error.

Reply:
Good catch. Done.